docs/cpp.rst

```diff
@@ -0,0 +1,68 @@
+C++ domain customization
+========================
+
+.. confval:: cpp_generate_synopses
+
+   :python:`bool` specifying whether to generate a *synopsis* for C++ domain
+   objects based on the first paragraph of their content (first sentence for
+   parameters).  The synopsis is shown as a tooltip when hovering over a
+   cross-reference link, and is also shown in the search results list.
+
+   Defaults to :python:`True`.
+
```

Comment:
It would be harder to add an example here because the user needs to be instructed where to hover their mouse cursor. Again, maybe a link to an example description that is worded like
> This sentence is part of the tooltip in the right side ToC. Hover your mouse cursor over this member's entry in the right side ToC to see it.

Reply:
I added an example.